packages/amplify-category-api/src/provider-utils/awscloudformation/utils/getAppSyncApiName.ts

```diff
@@ -0,0 +1,15 @@
+import { $TSContext } from 'amplify-cli-core';
+
+export const getAppSyncApiResourceName = async (context: $TSContext): Promise<string> => {
+  const { allResources } = await context.amplify.getResourceStatus();
+  const apiResource = allResources.filter((resource: { service: string }) => resource.service === 'AppSync');
+  let apiResourceName;
+
+  if (apiResource.length > 0) {
```

Comment:
please take care when apiResource is null

Reply:
apiResource is a array , I am checking the length
above line will return an empty array